client/login/magic-login/magic-login-email/magic-login-email-icon.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { useState, useEffect } from 'react';
+
+interface MagicLoginEmailIconProps {
+	icon: string;
+}
+
+export function MagicLoginEmailIcon( { icon }: MagicLoginEmailIconProps ) {
+	const [ IconComponent, setIconComponent ] = useState< React.ReactNode | null >( null );
+
+	useEffect( () => {
+		const getIconComponent = async ( iconName: string ) => {
+			try {
+				// Use dynamic import to load the icons.
+				const module = await import( `calypso/components/social-icons/${ iconName }` );
+				const IconComponent = module.default;
+				setIconComponent( <IconComponent /> );
+			} catch ( error ) {
+				setIconComponent( null );
+			}
+		};
+
+		getIconComponent( icon );
+	}, [ icon ] );
+
+	return <>{ IconComponent }</>;
+}
```

Comment:
You can use `AsyncLoad` component for this. It does this automatically.

Reply:
Ahh this is great! Thank you! 👍